pkg/master/reconcilers/endpointsadapter.go

```diff
@@ -69,18 +70,21 @@ func (adapter *EndpointsAdapter) Create(namespace string, endpoints *corev1.Endp
 // updated. The updated Endpoints object or an error will be returned.
 func (adapter *EndpointsAdapter) Update(namespace string, endpoints *corev1.Endpoints) (*corev1.Endpoints, error) {
 	endpoints, err := adapter.endpointClient.Endpoints(namespace).Update(endpoints)
-	if err == nil && adapter.endpointSliceClient != nil {
-		_, err = adapter.ensureEndpointSliceFromEndpoints(namespace, endpoints)
+	if err == nil {
+		_, err = adapter.EnsureEndpointSliceFromEndpoints(namespace, endpoints)
 	}
 	return endpoints, err
 }
 
-// ensureEndpointSliceFromEndpoints accepts a namespace and Endpoints resource
-// and creates or updates a corresponding EndpointSlice. The EndpointSlice
-// and/or an error will be returned.
-func (adapter *EndpointsAdapter) ensureEndpointSliceFromEndpoints(namespace string, endpoints *corev1.Endpoints) (*discovery.EndpointSlice, error) {
+// EnsureEndpointSliceFromEndpoints accepts a namespace and Endpoints resource
+// and creates or updates a corresponding EndpointSlice if an endpointSliceClient
+// exists. The EndpointSlice and/or an error will be returned.
+func (adapter *EndpointsAdapter) EnsureEndpointSliceFromEndpoints(namespace string, endpoints *corev1.Endpoints) (*discovery.EndpointSlice, error) {
```

Comment:
the `*discovery.EndpointSlice` return value is never used, and can be nil if the endpointsslice client is not set. I'd suggest omitting the return value.

Reply:
Done